### PR TITLE
fix(lib-injection): publish release tag of image

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Currently, only dev images for commits are pushed to the container
registry. This means that images like `v1.7.0` that get published are
really just aliases of the corresponding SHA image. These images use

```
pip install git+....@sha
```
to install the image.

For example:

```
~/d/h/c/datadog ❯❯❯ docker run -it gcr.io/datadoghq/dd-lib-python-init:v1.7.0 cat /datadog-init/sitecustomize.py | head -n 12
"""
This module when included on the PYTHONPATH will install the ddtrace package from pypi
for the Python runtime being used.
"""
$ docker run -it gcr.io/datadoghq/dd-lib-python-init:v1.7.0 cat /datadog-init/sitecustomize.py | head -n 12
...
version = "git+https://github.com/Datadog/dd-trace-py@ced65df45a9b592dd36af12c37c047c1486af38a"
```

While this isn't _wrong_ it is slow for auto instrumented applications
to start since the install is from source. It also means all
installation dependencies are required and the install could fail.

The solution is to run the publishing Github workflow on release after
the pypi_upload job. This should ensure that we build an image that uses
the release tag and subsequently pip installs with that version rather
than from source.

The Gitlab job responsible for publishing the image to the other
container registries is updated to use the release tag image rather than
the SHA. There is and was a race between when the Github registry image
is published and when the Gitlab job runs. To mitigate this the Gitlab
job is delayed by 1 day to give time for the image to be pushed to the
Github registry.

## Resources

- https://docs.gitlab.com/ee/ci/jobs/job_control.html#run-a-job-after-a-delay

### CODE REVIEW

**Comment by Kyle-Verhoog** [comment]
🤔 I'm not seeing the dev workflow run, gotta look into this

**Review by P403n1x87 — DISMISSED** [review]

File: .github/workflows/build_deploy.yml
Comment (brettlangdon) [inline .github/workflows/build_deploy.yml:216]: we also have the wheels here too if you want to use the local ones for building the image.

File: .github/workflows/build_deploy.yml
Comment (brettlangdon) [inline .github/workflows/build_deploy.yml:218]: this won't run unless the conditions that causes `uplod_pypi` runs right?

File: releasenotes/notes/lib-injection-0041b50a263e1c19.yaml
Comment (brettlangdon) [inline releasenotes/notes/lib-injection-0041b50a263e1c19.yaml:4]: do we need this? 🤔 

since this is a workflow/build change.... but it does impact the library images which are version specific...

Comment (Yun-Kim) [reply releasenotes/notes/lib-injection-0041b50a263e1c19.yaml:4]: I think we do need this, since technically this is a user-facing change by speeding up installs

**Review by Yun-Kim — APPROVED** [review]

File: .github/workflows/build_deploy.yml
Comment (brettlangdon) [reply .github/workflows/build_deploy.yml:218]: hmm, there isn't any way to verify this until we merge right?

File: .github/workflows/build_deploy.yml
Comment (mabdinur) [inline .github/workflows/build_deploy.yml:233]: Why are we publishing images for only these platforms? Do we plan to add more?

File: .github/workflows/build_deploy.yml
Comment (mabdinur) [inline .github/workflows/build_deploy.yml:234]: github.ref_name will reference the tag in a release. In this case would this resolve to `DDTRACE_PYTHON_VERSION=vv{release}`? Is this correct?

File: .gitlab-ci.yml
Comment (mabdinur) [inline .gitlab-ci.yml:60]: What's the impact when this workflow fails? Who gets notified? Would library injection fail or would the library be installed from source?

File: .github/workflows/build_deploy.yml
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:234]: great catch! yes since this should run with the tag that is published so it would result in the double v.

File: .github/workflows/build_deploy.yml
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:224]: nope! just some copy-paste

File: .github/workflows/build_deploy.yml
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:233]: this isn't new or changing, we only plan to support these for now

File: .github/workflows/build_deploy.yml
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:216]: this is out of scope of this change

File: .github/workflows/build_deploy.yml
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:218]: there isn't, but it's effectively the same job that runs on commit to build the test image (only difference is the tag)

File: .gitlab-ci.yml
Comment (Kyle-Verhoog) [reply .gitlab-ci.yml:60]: This is a good question. AFAIK we don't have good visibility of any of the gitlab jobs. We should look into how we can monitor the jobs.

Comment (Kyle-Verhoog) [reply .gitlab-ci.yml:60]: Looks like we have the results in CI app. I will set up alerting if the pipeline fails.

File: .github/workflows/build_deploy.yml
Comment (brettlangdon) [reply .github/workflows/build_deploy.yml:216]: yup, just a thought I had

**Review by mabdinur — DISMISSED** [review]

**Review by mabdinur — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**
